tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/AWS/SDK/IRequest.cs

```diff
@@ -17,12 +17,12 @@ internal interface IRequest
         /// <summary>
         /// Gets the service endpoint
         /// </summary>
-        Uri? Endpoint { get; }
+        Uri Endpoint { get; }
```

Comment:
cc @andrewlock @duncanista This is a case where I'm explicitly saying that all AWS request will always have non-null values for these fields. I think this is safe to do. Do you agree?

Reply:
By looking at the v1 and v3 docs, it looks like there will always be an `Endpoint` as per [v1 docs](https://docs.aws.amazon.com/sdkfornet1/latest/apidocs/html/T_Amazon_Runtime_Internal_IRequest.htm) (not available for inspection in v3). Where the only mention I could find was [this](https://docs.aws.amazon.com/sdkfornet/v3/apidocs/items/Runtime/TIRequestContext.html). It should be safe, our integration tests should cover this, since they mock the service, but they still bring up a server where an endpoint is being hit.

Reply:
Thanks, even in our integration tests against localstack I noticed that we always have a `http.url` span tag, so I used that as evidence that this would be okay 😄